src/s2/s2polyline.cc

```diff
@@ -50,6 +50,13 @@ static const unsigned char kCurrentLosslessEncodingVersionNumber = 1;
 S2Polyline::S2Polyline()
   : s2debug_override_(S2Debug::ALLOW) {}
 
+S2Polyline::S2Polyline(S2Polyline&& other)
+ : s2debug_override_(other.s2debug_override_),
+   num_vertices_(other.num_vertices_),
+   vertices_(std::move(other.vertices_)) {
+  other.num_vertices_ = 0;
```

Comment:
It might be safer/more obvious to do `other_.vertices_.clear()` to get back to the normal empty state.

Reply:
I don't think it's necessary to add a `release()` operation here. It's explicit in the standard that moving a `unique_ptr` has the same effect as calling `release()` on the source operand.